chaosazure/vmss/actions.py

```diff
@@ -242,9 +248,38 @@ def choose_vmss_instance_at_random(vmss_choice, configuration, secrets):
         logger.debug(
             "Found virtual machine scale set instances: {}".format(
                 [x['name'] for x in vmss_instances]))
-    choice_vmss_instance = random.choice(vmss_instances)
+    choice_vmss_instance = random.choice(vmss_instance)
+    return choice_vmss_instance
+
+def choose_vmss_instance(vmss_choice, configuration, secrets, instance_criteria):
+    vmss_instances = fetch_vmss_instances(vmss_choice, configuration, secrets)
+    if not vmss_instances:
+        logger.warning("No virtual machine scale set instances found")
+        raise FailedActivity("No virtual machine scale set instances found")
+    else:
+        logger.debug(
+            "Found virtual machine scale set instances: {}".format(
+                [x['name'] for x in vmss_instances]))
+    logger.warning("Attempting to stop instance that matches one of the following criteria %s" % instance_criteria)
+    choice_vmss_instance = next(vmss for vmss in vmss_instances if vmss_matches_criteria(vmss, instance_criteria))
+    logger.warning("Attempting to stop instance with name %s" % choice_vmss_instance['name'])
+    
     return choice_vmss_instance
 
+def vmss_matches_criteria(vmss, instance_criteria):
+    for criteriaMap in instance_criteria:
+        logger.debug("Checking criteriaMap %s" % criteriaMap)
+        foundMismatch = False
+        for key, value in criteriaMap.items():
```

Comment:
same remark on `criteriaMap` vs `criteria_map`.

Also, while Python is not statically types, we don't tend to specify the type in the name. So `criterias` might be better than `criteriaMap`.

Should we drop Python 3.5 someday, we could actually add the typing annotation to variable declaration: https://www.python.org/dev/peps/pep-0526/